plugins/hosts/linux/cap/nfs.rb

```diff
@@ -114,13 +90,50 @@ def self.nfs_opts_setup(folders)
 
             opts[:linux__nfs_options] << "anonuid=#{opts[:map_uid]}" if !hasuid
             opts[:linux__nfs_options] << "anongid=#{opts[:map_gid]}" if !hasgid
-            opts[:linux__nfs_options] << "fsid=#{opts[:uuid]}"
+            opts[:linux__nfs_options] << "fsid=#{opts[:fsid]}"
+            # We need to store reference to machine somewhere, `replicas`
+            # seems to be the only place it is possible and harmless
+            opts[:linux__nfs_options] << "replicas=#{opts[:uuid]}"
           end
         end
 
         def self.nfs_running?(check_command)
           system(check_command)
         end
+
+        def self.nfs_current_exports
+          exports = `sudo exportfs -v`
+          # exportfs output should be properly parsed
+          #
+          # This is a little bit complicated because folder with long path looks like this:
+          #   `sudo exportfs -v`
+          #   #=> "/very_long_path\n\t\t192.168.122.4(rw,all_squash,no_subtree_check,fsid=180077614,replicas=8c085523-3e67-48b3-a362-613167dde240)"
+          #
+          # while short path looks like this:
+          #   `sudo exportfs -v`
+          #   #=> "/short_path      \t192.168.122.153(rw,all_squash,no_subtree_check,fsid=180077614,replicas=8c085523-3e67-48b3-a362-613167dde240)"
+          #
+          # We need to properly handle both cases
+          exports = exports.split("\n")      # each new line is either folder or ip
+          exports.map! { |e| e.split("\t") } # tab prepends ip
+          exports.flatten!                   # make one-dimensional array
+          exports.select! { |e| !e.empty? }  # remove all empty strings created by splitting with long entries
+          exports.map!(&:strip)              # make sure there are no leading spaces
+
+          folders = exports.each_slice(2).map(&:first) # array of host paths
+          opts    = exports.each_slice(2).map(&:last)  # array of IP address and options
```

Comment:
This whole block of code seems incredibly fragile to me. Can we use a regular expression or cleaner function instead? Thank you for putting comments, but in my experience, if you need this many comments to explain a Ruby function, it's an indication that it needs a refactor.